src/tools/rpc.rs

```diff
@@ -0,0 +1,224 @@
+///
+/// When a client first joins the network or has fallen behind and needs to catch
+/// up, certificates allow the client to verify that a block someone gives them
+/// is the real one.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Certificate {
+    /// Proposal value.
+    #[serde(default)]
+    pub prop: Option<CertificateProposal>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct CertificateProposal {
+    /// Block header's hash.
+    #[serde(rename = "dig")]
+    pub block_digest: HashDigest,
+}
+
+/// BlockHeader
+/// Deserialized from MessagePack format.
+///
+/// See [block.go](https://github.com/algorand/go-algorand/blob/master/data/bookkeeping/block.go) for more details.
+// Comments below are simply copied from the go-algorand repo.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BlockHeaderMsgPack {
+    /// RewardsLevel specifies how many rewards, in MicroAlgos, have been distributed
+    /// to each config.Protocol.RewardUnit of MicroAlgos since genesis.
+    #[serde(default)]
+    pub earn: u64,
+
+    /// The FeeSink accepts transaction fees. It can only spend to the incentive pool.
+    #[serde(default)]
+    pub fees: Option<HashDigest>,
+
+    /// The number of leftover MicroAlgos after the distribution of RewardsRate/rewardUnits
+    /// MicroAlgos for every reward unit in the next round.
+    #[serde(default)]
+    pub frac: u64,
+
+    /// Genesis ID to which this block belongs.
+    #[serde(default)]
+    pub gen: String,
+
+    /// Genesis hash to which this block belongs.
+    #[serde(default)]
+    pub gh: Option<HashDigest>,
+
+    /// The hash of the previous block.
+    #[serde(default)]
+    pub prev: Option<HashDigest>,
+
+    /// Current protocol.
+    #[serde(default)]
+    pub proto: String,
+
+    /// The number of new MicroAlgos added to the participation stake from rewards at the next round.
+    #[serde(default)]
+    pub rate: u64,
+
+    /// Round represents a protocol round index.
+    #[serde(default)]
+    pub rnd: u64,
+
+    /// The round at which the RewardsRate will be recalculated.
+    #[serde(default)]
+    pub rwcalr: u64,
+
+    /// The RewardsPool accepts periodic injections from the FeeSink and continually
+    /// redistributes them to addresses as rewards.
+    #[serde(default)]
+    pub rwd: Option<HashDigest>,
+
+    /// Sortition seed.
+    #[serde(default, deserialize_with = "deserialize_byte32_arr_opt")]
+    pub seed: Option<[u8; 32]>,
+
+    /// TimeStamp in seconds since epoch.
+    #[serde(default)]
+    pub ts: i64,
+
+    /// Root of transaction merkle tree using SHA512_256 hash function.
+    /// This commitment is computed based on the PaysetCommit type specified in the block's consensus protocol.
+    #[serde(default)]
+    pub txn: Option<HashDigest>,
+
+    /// Root of transaction vector commitment merkle tree using SHA256 hash function.
+    #[serde(default)]
+    pub txn256: Option<HashDigest>,
+}
+
```

Comment:
For the below part, I used algonaut code base as a reference. This [commit](https://github.com/manuelmauro/algonaut/commit/3ceaecc056b5b4fe9da3452c9023236195e932d6), to be specific. I might send 'thanks' to those guys 🙂